networkit/cpp/generators/RmatGenerator.cpp

```diff
@@ -3,26 +3,143 @@
+
+    assert((size & (size - 1)) == 0); // The size must be a power of two.
+    std::vector<Entry> entryList;
+    generateEntries(entryList, size, a, b, c, d);
+    while (entryList.size() < size) { // Fill the remaining entries with 0-probability entries.
+        entryList.push_back(Entry{0, 0, 0, 0});
+    }
+
+    // Construct the alias table:
+    mask = size - 1;
+    bits.resize(size);
+    numberOfBits.resize(size);
+    coinFlipProbability.resize(size);
+    coinFlipReplacement.resize(size);
+    for (count i = 0; i < size; i++) {
+        bits[i] = std::make_pair(entryList[i].i, entryList[i].j);
+        numberOfBits[i] = entryList[i].numberOfBits;
+        coinFlipProbability[i] = 0;
+        coinFlipReplacement[i] = 0;
+    }
+    double baseProbability = 1.0 / static_cast<double>(size);
+    count lastOverfullIndex = 0;
+    count lastUnderfullIndex = 0;
+    do {
+        while (entryList[lastUnderfullIndex].probability >= baseProbability) {
+            lastUnderfullIndex++;
+            if (lastUnderfullIndex == size)
+                return;
+        }
+        int curUnderfullIndex = lastUnderfullIndex;
+        do {
+            while (entryList[lastOverfullIndex].probability <= baseProbability) {
+                lastOverfullIndex++;
+                if (lastOverfullIndex == size)
+                    return;
+            }
+            double delta = baseProbability - entryList[curUnderfullIndex].probability;
+            entryList[curUnderfullIndex].probability = baseProbability;
+            entryList[lastOverfullIndex].probability -= delta;
+            coinFlipReplacement[curUnderfullIndex] = lastOverfullIndex;
+            coinFlipProbability[curUnderfullIndex] = static_cast<uint32_t>(
+                delta / baseProbability * std::numeric_limits<uint32_t>::max());
+            if (entryList[lastOverfullIndex].probability < baseProbability
+                && lastOverfullIndex < lastUnderfullIndex) {
+                curUnderfullIndex = lastOverfullIndex;
+            } else
+                break;
+        } while (true);
+    } while (true);
+}
+
+std::pair<node, node> RmatGenerator::sampleEdge(uint8_t input_bits) {
+    std::pair<node, node> result{0, 0};
+    do {
+        if (remainingBits >= input_bits) {
+            remainingBits -= input_bits;
+            result.first = result.first << input_bits | curBits.first >> remainingBits;
+            result.second = result.second << input_bits | curBits.second >> remainingBits;
+            curBits.first &= (1 << remainingBits) - 1;
+            curBits.second &= (1 << remainingBits) - 1;
+            return result;
+        }
+        result.first = result.first << remainingBits | curBits.first;
+        result.second = result.second << remainingBits | curBits.second;
+        input_bits -= remainingBits;
+
+        // sample step
+        [this]() {
+            uint64_t randomNumber = Aux::Random::integer();
```

Comment:
Avoid `Aux::Random::integer();`, since this function causes a slowdown due to every time creating a new distribution.